src/pal/tools/gen-buildsys-gcc.sh

```diff
@@ -159,7 +161,7 @@ fi
 overridefile=gcc-compiler-override.txt
 
 # Determine the current script directory
-__currentScriptDir="$(cd -- "$(dirname -- "$0")" && pwd -P)"
+__currentScriptDir="$script_dir"
```

Comment:
nit: since now it is not determining the path, we can remove the comment and this line by inlining `$script_dir` at line 168.